src/biogeochem/CropType.F90

```diff
@@ -569,14 +584,22 @@ subroutine CropUpdateAccVars(this, bounds, t_ref2m_patch, t_soisno_col)
        write(iulog,*)'update_accum_hist allocation error for rbuf1dp'
        call endrun(msg=errMsg(sourcefile, __LINE__))
     endif
+    allocate(rbufslp2(begp:endp), stat=ier)
+    if (ier/=0) then
+       write(iulog,*)'update_accum_hist allocation error for rbuf1dp (2)'
+       call endrun(msg=errMsg(sourcefile, __LINE__))
+    endif
 
-    ! Accumulate and extract GDDPLANT
+    ! Accumulate and extract HUI and GDDACCUM
     
-    call extract_accum_field ('GDDPLANT', rbufslp, nstep)
+    call extract_accum_field ('HUI', rbufslp, nstep)
+    call extract_accum_field ('GDDACCUM', rbufslp2, nstep)
     do p = begp,endp
-      rbufslp(p) = max(0.0_r8,this%gddplant_patch(p)-rbufslp(p))
+      rbufslp(p) = max(0.0_r8,this%hui_patch(p)-rbufslp(p))
+      rbufslp2(p) = max(0.0_r8,this%gddaccum_patch(p)-rbufslp2(p))
     end do
-    call update_accum_field  ('GDDPLANT', rbufslp, nstep)
+    call update_accum_field  ('HUI', rbufslp, nstep)
+    call update_accum_field  ('GDDACCUM', rbufslp2, nstep)
```

Comment:
- [ ] Is this initial extract/modify/update step needed for GDDACCUM? I think that, if you look for other uses of accumulation fields, you'll find that this is a non-standard thing (GDDTSOI below is a decent example). I'm thinking that it is needed for HUI because of the possible update to this variable that motivated #1655 , but without an update like that, you don't need this extra step. You should double-check me by looking at other uses of accumulation fields and/or removing this code for GDDACCUM and verifying that answers are the same either way.

- [ ] [OPTIONAL] If you make this change, then I think you could safely remove the rbufslp2 variable, just using rbufslp directly in the update_accum_field for GDDACCUM below.